lib/phoenix_pubsub/tracker.ex

```diff
@@ -224,6 +224,15 @@ defmodule Phoenix.Tracker do
     permdown_period      = opts[:permdown_period] || 1_200_000
     clock_sample_periods = opts[:clock_sample_periods] || 2
     log_level            = Keyword.get(opts, :log_level, false)
+
+    if down_period <= (2 * broadcast_period) do
+      raise ArgumentError, message: "down_period must be at least twice as large as the broadcast_period"
+    end
+
+    if permdown_period <= down_period do
+      raise ArgumentError, message: "permdown_period must be at least larger than the down_period"
+    end
+
```

Comment:
let's take inspiration from José's tweet:
https://github.com/elixir-lang/gen_router/blob/2001c015b560d82f8077f323de5ed0103f5e0566/lib/dynamic_supervisor.ex#L398-L404

Since we are Elixir 1.2 now, we can use start using `with`